src/thunderbots/software/test/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -109,3 +109,29 @@ TEST(TacticTest, test_tactic_reports_done_at_same_time_nullptr_returned)
     EXPECT_FALSE(intent_ptr);
     EXPECT_TRUE(tactic.done());
 }
+
+TEST(TacticTest, test_tactic_restarts_when_set_to_loop_infinitely)
+{
+    Robot robot = Robot(0, Point(), Vector(), Angle::zero(), AngularVelocity::zero(),
+                        Timestamp::fromSeconds(0));
+
+    // Create a MoveTactic that wants to move the Robot to where it already is.
+    // Therefore we expect the tactic to be done
+    MoveTactic tactic = MoveTactic(true);
+    tactic.updateRobot(robot);
+    tactic.updateParams(Point(), Angle::zero(), 0.0);
+
+    // Since the Tactic should be done, we expect the returned intent_ptr to alternate
+    // between some valid Intent, and nullptr as the Tactic finishes and restarts
```

Comment:
Wait, why? If we're infinitely looping, we would expected to never recieve any indication that Tactic has restarted?

Reply:
Fixed this so that we restart the coroutine and take the new result. nullptr will now never be returned if we are looping.